src/librustdoc/html/render/cache.rs

```diff
@@ -251,6 +252,19 @@ fn get_index_type_name(clean_type: &clean::Type, accept_generic: bool) -> Option
     }
 }
 
+fn get_generics(clean_type: &clean::Type) -> Option<Vec<Generic>> {
+    clean_type.generics().and_then(|types| {
+        let r = types
+            .iter()
+            .filter_map(|t| {
+                get_index_type_name(t, false)
+                    .map(|name| Generic { name: name.as_str().to_ascii_lowercase() })
+            })
+            .collect::<Vec<_>>();
```

Comment:
What happens in case there is the same type used in generics multiple times:

```
fn foo<T: Debug, U: Debug, Y: Debug>() {}
```

Will `Debug` be added 3 times?

Reply:
Yeah, it does. And the JavaScript search tool actually does take advantage of this. You can see in the search code how it's counting the number of times a generic appears in the search query and in the type-being-checked:

https://github.com/rust-lang/rust/blob/4169b402bb6e5876d7dc5f8cf8b31def451badff/src/librustdoc/html/static/search.js#L361-L383

If you mention Debug twice in your query, then it must appear twice in the type signature.

Reply:
Oh I see. Please add a comment about this behaviour then so the next ones reading this code won't ask the same question.

Reply:
Please also add a test to check this behaviour (regressions did a lot of bad on the search). I think after that it'll be good. Just need to run the perf check and we're good to go. Thanks a lot for fixing this issue!

Reply:
Okay, I added a test case for "bag semantics" search results. It also required a bug fix for the query parsing, since it was breaking up `Foo<A, B>, Bar` into three queries, when it needed to be two.